cpp/src/Glacier2Lib/NullPermissionsVerifier.cpp

```diff
@@ -26,114 +26,88 @@ class NullSSLPermissionsVerifier : public Glacier2::SSLPermissionsVerifier
 {
 public:
 
-    virtual bool
-    authorize(Glacier2::SSLInfo, string&, const Ice::Current&) const
+    bool authorize(Glacier2::SSLInfo, string&, const Ice::Current&) const final
     {
         return true;
     }
 };
 
-class Init
+void setupObjectAdapter(
+    const Ice::CommunicatorPtr& communicator,
+    Ice::ObjectAdapterPtr& adapter)
 {
-public:
-
-    Init(const CommunicatorPtr&, const string&, const vector<string>&);
-
-private:
-
-    string checkPermissionVerifier(const string&);
-    void createObjects();
+    if (adapter == nullptr)
+    {
+        // Create collocated object adapter
+        adapter = communicator->createObjectAdapter("");
+        adapter->activate();
+    }
+}
 
-    const CommunicatorPtr _communicator;
-    ObjectAdapterPtr _adapter;
+}
 
-    Identity _nullPVId;
-    Identity _nullSSLPVId;
-};
+namespace Glacier2Internal
+{
 
-Init::Init(const CommunicatorPtr& communicator, const string& category, const vector<string>& props) :
-    _communicator(communicator)
+void
```

Comment:
Why is one function in the anonymous namespace and the other in Glacier2Internal?
If it's the definition of a function declared in some header, it would be nicer to just define it with

void
Glacier2Internal::...

Reply:
The one in the anonymous namespace is only used here, `setupNullPermissionsVerifier` in Glacier2Internal is used by Glacier2Router and IceGrid to setup this permissions verifier.

Reply:
I removed the helper method that setup the adapter